src/lib/nvme_pt_ioctl.c

```diff
@@ -727,6 +733,94 @@ static int validate_sessn(struct opal_device *dev)
 	return 0;
 }
 
+static int  parse_tper_host_prop(struct sed_device *device)
+{
+	struct opal_device *dev = device->priv;
+	int payload_len = dev->payload.len;
+	struct sed_tper_properties *tper = &device->discv.sed_tper_props;
+	uint64_t val_ptr = sizeof(struct sed_tper_properties);
+	uint64_t val, data_ptr = 0;
+
+	for (int i = 0; i < payload_len; i++) {
+		if(resp_token_match(dev->payload.tokens[i], OPAL_STARTNAME)) {
+			val = dev->payload.tokens[i+2]->vals.uint;
+
+			if (data_ptr < val_ptr)  {
+				memcpy((uint8_t *)tper + data_ptr, &val, sizeof(uint64_t));
```

Comment:
Properties are returned as key-value pairs. there needs to be code that matches key (property name) to a specific member in properties data structure.